internal/credential/state.go

```diff
@@ -192,7 +204,7 @@ func (s *AwsCredentialPersistedState) GenerateCredentialChain(ctx context.Contex
 // return a map for long-term credentials with following keys:
 // access_key_id, secret_access_key & creds_last_rotated_time
 func (s *AwsCredentialPersistedState) ToMap() map[string]any {
-	if HasDynamicCredentials(s.CredentialsConfig.AccessKey) {
+	if GetCredentialType(s.CredentialsConfig) == DynamicAWS {
```

Comment:
Do you think we should also return an empty map for unknown credential types? I'm not sure if there is a use case for s3 compatible services.

Reply:
> Do you think we should also return an empty map for unknown credential types?

I think so - I've made this change as well https://github.com/hashicorp/boundary-plugin-aws/pull/33/commits/d1a59dbd8aa3a3c2169e60b3a2b7ea3d336e0762

In the case of s3-compatible services, we still need to return the actual `access_key_id` and `secret_access_key` back to Boundary for storage. This func gets called when we build the proto for response back to Boundary, so if we carved it out, I think we wouldn't be able to do any subsequent calls as they wouldn't have the credentials.